src/Promitor.Integrations.Azure/Authentication/AzureAuthenticationFactory.cs

```diff
@@ -69,24 +69,26 @@ public static AzureAuthenticationInfo GetConfiguredAzureAuthentication(IConfigur
         /// <summary>
         ///     Gets a valid token using a Service Principal or a Managed Identity
         /// </summary>
-        public static async Task<TokenCredentials> GetTokenCredentialsAsync(string resource, string tenantId, AzureAuthenticationInfo authenticationInfo)
+        public static async Task<TokenCredentials> GetTokenCredentialsAsync(string resource, string tenantId, AzureAuthenticationInfo authenticationInfo, System.Uri azureAuthorityHost)
         {
             Guard.NotNullOrWhitespace(resource, nameof(resource));
             Guard.NotNullOrWhitespace(tenantId, nameof(tenantId));
             Guard.NotNull(authenticationInfo, nameof(authenticationInfo));
 
             TokenCredential tokenCredential;
 
+            var tokenCredentialOptions = new TokenCredentialOptions { AuthorityHost = azureAuthorityHost };
+
             switch (authenticationInfo.Mode)
             {
                 case AuthenticationMode.ServicePrincipal:
-                    tokenCredential = new ClientSecretCredential(tenantId, authenticationInfo.IdentityId, authenticationInfo.Secret);
+                    tokenCredential = new ClientSecretCredential(tenantId, authenticationInfo.IdentityId, authenticationInfo.Secret, tokenCredentialOptions);
                     break;
                 case AuthenticationMode.UserAssignedManagedIdentity:
-                    tokenCredential = new ManagedIdentityCredential(authenticationInfo.IdentityId);
+                    tokenCredential = new ManagedIdentityCredential(authenticationInfo.IdentityId, tokenCredentialOptions);
                     break;
                 case AuthenticationMode.SystemAssignedManagedIdentity:
-                    tokenCredential = new ManagedIdentityCredential();
+                    tokenCredential = new ManagedIdentityCredential( "",tokenCredentialOptions);
```

Comment:
Are we sure this is OK and won't blow up?

Reply:
Ahh I think I made a mistake here, should be

```
tokenCredential = new ManagedIdentityCredential(options:tokenCredentialOptions);
```
or 
```
tokenCredential = new ManagedIdentityCredential( default ,tokenCredentialOptions);
```

Reply:
Let's go for new ManagedIdentityCredential(options:tokenCredentialOptions); maybe?